pkgs/development/python-modules/dufte/default.nix

```
@@ -0,0 +1,42 @@
+{ stdenv
+, buildPythonPackage
+, fetchPypi
+, isPy3k
+, importlib-metadata
+, matplotlib
+, numpy
+, pytestCheckHook
+}:
+
+buildPythonPackage rec {
+  pname = "dufte";
+  version = "0.2.9";
+  disabled = !isPy3k;
+
+  src = fetchPypi {
+    inherit pname version;
+    sha256 = "0nkaczipbsm8c14j9svxry2wigmn5iharibb6b8g062sjaph8x17";
+  };
+  format = "pyproject";
+
+  propagatedBuildInputs = [
+    importlib-metadata
+    matplotlib
+    numpy
+  ];
```

https://github.com/nschloe/dufte/blob/b08815ea12c6dbff419ff65aa36860f2af54afaf/setup.cfg#L30
```suggestion
  propagatedBuildInputs = [
    matplotlib
    numpy
  ] ++ lib.optionals (pythonOlder "3.8") [
    importlib-metadata
  ];
```

---

Did we ever consider making these "backport" libraries simply evaluate to null (or something else harmless/empty) on a non-applicable python version?

---

a lot of them do
```
[10:13:08] jon@nixos ~/projects/nixpkgs/pkgs/development/python-modules (backport-python-dask)
$ rg "then null"
enum34/default.nix
8:if pythonAtLeast "3.4" then null else buildPythonPackage rec {

ipaddress/default.nix
8:if (pythonAtLeast "3.3") then null else buildPythonPackage rec {

backports_shutil_get_terminal_size/default.nix
8:if !(pythonOlder "3.3") then null else buildPythonPackage {

cffi/default.nix
3:if isPyPy then null else buildPythonPackage rec {

functools32/default.nix
7:if isPy3k then null else buildPythonPackage rec {

backports_lzma/default.nix
10:if !(pythonOlder "3.3") then null else buildPythonPackage rec {
```
but it still causes failure for some https://github.com/NixOS/nixpkgs/pull/100017

---

see https://github.com/rytilahti/python-miio/pull/828 as an example